sts_synindex_external.R

```diff
@@ -60,16 +61,17 @@ duplicate_folder(source_folder = PARQUET_FILTERED_LOCATION,
                  destination_folder = PARQUET_FINAL_LOCATION)
 
 copy_folders_reparent <- function(source_folder, destination_folder) {
-  folders_to_copy <- setdiff(list.dirs(source_folder, recursive = F, full.names = F), 
-                             list.dirs(destination_folder, recursive = F, full.names = F))
+  folders_to_copy <- 
```

Comment:
Second note, please look into this package: https://testthat.r-lib.org. We should try to unit test as much of our code as possible.

Unfortunately, I'm not really an expert when it comes to writing R tests and the ability to mock API calls.

Reply:
Thanks for the suggestion. I've only previously used `testthat` for package development, so I didn't think to use it for non-package development/testing. I will use `testthat` (instead of manual testing in the console) moving forward for non-package development.

Reply:
Since the code touches data that requires FISMA compliance - it's important that any code we write is tested.

Reply:
I agree. For non-package code development, I have been testing/debugging in the console or writing tests to temporary files, instead of writing automated tests to a separate test file/directory.

Similar to `roxygen`, I was introduced to `testthat` when first learning how to build R packages, so I thought to use `testthat` for only package development, not for non-package R code as well.

Reply:
[bdc670d](https://github.com/Sage-Bionetworks/recover-parquet-external/pull/4/commits/bdc670d6e18f367aa1b2c7c0c46c20c7e401536a)